Cargo.toml

```diff
@@ -1,37 +1,12 @@
-[package]
-name = "taiga"
-version = "0.1.0"
-edition = "2021"
-
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
-
-[dependencies]
-ark-serialize = { version = "^0.3.0", default-features = false }
-ark-ff = { git="https://github.com/simonmasson/algebra", rev="e2ea75c" }
-ark-ec = { git="https://github.com/simonmasson/algebra", rev="e2ea75c" }
-ark-poly-commit = "0.3"
-ark-poly = "0.3"
-ark-bls12-381-new = { git="https://github.com/simonmasson/curves/", branch="new-pairing-layer" }
-ark-ed-on-bls12-381-new = { git="https://github.com/simonmasson/curves/", branch="new-pairing-layer" }
-ark-bw6-764-new = { git="https://github.com/simonmasson/curves/", branch="new-pairing-layer" }
-
-rand = "0.8.4"
-sha2 = "0.10.1"
-plonk-core = { git = "https://github.com/heliaxdev/ark-plonk/", features = ["trace"], branch="taiga/randomized-circuits-rebase" }
-plonk-hashing = { git = "https://github.com/heliaxdev/ark-plonk/", branch="taiga/randomized-circuits-rebase" }
-blake2 = "0.9"
-rs_merkle = { git = "https://github.com/heliaxdev/rs-merkle/" } # "0.2.2"
-lazy_static = "1"
-derivative = { version = "2", features = ["use_core"] }
-thiserror = "1"
-blake2b_simd = "1"
-ark-std = "0.3"
-rand_core = {version = "0.6", default-features=false, features = ["getrandom"] }
-
-[dev-dependencies]
-ark-std = "0.3"
+[workspace]
+members = [
+    "taiga_halo2",
+    # "taiga_zk_garage",
+]
 
 [patch.crates-io]
-ark-serialize = { git="https://github.com/simonmasson/algebra", rev="e2ea75c" }
-ark-ff = { git="https://github.com/simonmasson/algebra", rev="e2ea75c" }
-ark-ec = { git="https://github.com/simonmasson/algebra", rev="e2ea75c" }
+# ark-serialize = { git="https://github.com/simonmasson/algebra", rev="e2ea75c" }
+# ark-ff = { git="https://github.com/simonmasson/algebra", rev="e2ea75c" }
+# ark-ec = { git="https://github.com/simonmasson/algebra", rev="e2ea75c" }
+halo2_gadgets = { git="https://github.com/XuyangSong/halo2", branch="taiga"}
```

Comment:
Could we folk the `halo2` to our heliaxdev repo?
I don't know if I have the permission

Reply:
Looks like it's already created, but I don't have permissions either - maybe @simonmasson does?

Reply:
I fetched with the upstream  the main branch of `heliaxdev/halo2`. From that, you can probably create do the modification as you did in your repo. NB: I created a branch `halo2` where I put the Poseidon4 parameters but we don't need it anymore because we use a different nullifier computation and there are only Poseidon2 functions now (no Poseidon4 or 8).